### ISSUE TITLE
vscode.workspace.findFiles cannot get files appeared in .gitignore

### ISSUE THREAD

**Issue by k8w** [Microsoft/vscode#36617]
- VSCode Version: 1.18.0-insider
- OS Version: win10

Steps to Reproduce:

```
|- lib
  |- Test.ts
|- .gitignore
```

.gitignore
```
/lib
```

```ts
vscode.workspace.findFiles('**/*.ts').then(files => {
    console.log(files)
})
```

**Expect**
Get Array[1] with `Test.ts` inside

**Actual**
Get Array[0]

Reproduces without extensions: Yes

**Comment by k8w** [comment]
This result in this: [https://github.com/pmneo/ts-importer/issues/45](https://github.com/pmneo/ts-importer/issues/45)

**Comment by roblourens** [comment]
@chrmarti I can repro this, which is very strange. But only with a pattern in gitignore that excludes a folder, like `/lib`. Not with `**/*.js`.

**Comment by chrmarti** [comment]
Fixed with 15449873.
